main/windows.py

```diff
@@ -56,9 +61,15 @@ def windows_convert_xlsx(excel, xlsx_filepath: str, pdf_filepath: str):
     :param pdf_filepath: path to future pdf file
     :type pdf_filepath: str
     """
-    xl_sheets = excel.Workbooks.Open(xlsx_filepath)
-    xl_sheets.Worksheets[0].ExportAsFixedFormat(0, pdf_filepath)
-    xl_sheets.Close(True)
+    sheets_opened = False
+    xl_sheets = None
+    try:
+        xl_sheets = excel.Workbooks.Open(xlsx_filepath)
+        sheets_opened = True
+        xl_sheets.Worksheets[0].ExportAsFixedFormat(0, pdf_filepath)
+    finally:
+        if sheets_opened and xl_sheets is not None:
+            xl_sheets.Close(True)
 
 
 def windows(paths, files, has_type_in_request: dict):
```

Comment:
types of `paths` and `files` are not specified

Reply:
bracnh a bit depricated